### PR TITLE
add limitation on some commissioner relative operations

### PR DESCRIPTION
**Pull request by librasungirl**

### CODE REVIEW

**Comment by librasungirl** [comment]
@GiedriusM we are now agreed to not allow AddJoiner() before the petition.
Feel free to tell us if you have any concerns. Thanks.

**Comment by GiedriusM** [comment]
#1486 for the reference. Can't think of any real problem with this right now, although it could be noted in the API. Other than that LGTM.

**Comment by jwhui** [comment]
Agree with @GiedriusM that the Doxygen should be updated to reflect these changes.

**Comment by librasungirl** [comment]
Updated. Thanks.

File: include/openthread/commissioner.h [hunk 1]
```diff
@@ -103,7 +107,7 @@ OTAPI ThreadError OTCALL otCommissionerRemoveJoiner(otInstance *aIntsance, const
  *
  * @param[in]  aProvisioningUrl  A pointer to the Provisioning URL (may be NULL).
  *
- * @retval kThreadError_None         Successfully added the Joiner.
+ * @retval kThreadError_None         Successfully set the Provisioning URL.
```
Comment (jwhui) [inline include/openthread/commissioner.h]: Should also document the new possible return values in this method and others.

File: include/openthread/commissioner.h [hunk 2]
```diff
@@ -112,14 +119,17 @@ OTAPI ThreadError OTCALL otCommissionerSetProvisioningUrl(otInstance *aInstance,
...
- * @param[in]  aAddress       A pointer to the IPv6 destination.
+ * @param[in]  aInstance             A pointer to an OpenThread instance.
+ * @param[in]  aChannelMask          The channel mask value.
+ * @param[in]  aCount                The number of energy measurements per channel.
+ * @param[in]  aPeriod               The time between energy measurements (milliseconds).
+ * @param[in]  aAddress              A pointer to the IPv6 destination.
+ *
+ * @retval kThreadError_None         Successfully enqueued the Announce Begin message.
+ * @retval kThreadError_NoBufs       Insufficient buffers to generate an Announce Begin message.
+ * @retval kThreadError_InvalidState The commissioner is not active
```
Comment (jwhui) [inline include/openthread/commissioner.h]: nit: add period (`.`) at end of line.

File: include/openthread/commissioner.h [hunk 3]
```diff
@@ -141,17 +151,20 @@ typedef void (OTCALL *otCommissionerEnergyReportCallback)(uint32_t aChannelMask,
...
+ * @param[in]  aCount                The number of energy measurements per channel.
+ * @param[in]  aPeriod               The time between energy measurements (milliseconds).
+ * @param[in]  aScanDuration         The scan duration for each energy measurement (milliseconds).
+ * @param[in]  aAddress              A pointer to the IPv6 destination.
+ * @param[in]  aCallback             A pointer to a function called on receiving an Energy Report message.
+ * @param[in]  aContext              A pointer to application-specific context.
+ *
+ * @retval kThreadError_None         Successfully enqueued the Energy Scan Query message.
+ * @retval kThreadError_NoBufs       Insufficient buffers to generate an Energy Scan Query message.
+ * @retval kThreadError_InvalidState The commissioner is not active
```
Comment (jwhui) [inline include/openthread/commissioner.h]: nit: add period (`.`) at end of line.

File: include/openthread/commissioner.h [hunk 4]
```diff
@@ -171,15 +184,18 @@ typedef void (OTCALL *otCommissionerPanIdConflictCallback)(uint16_t aPanId, uint
...
+ * @param[in]  aInstance             A pointer to an OpenThread instance.
+ * @param[in]  aPanId                The PAN ID to query.
+ * @param[in]  aChannelMask          The channel mask value.
+ * @param[in]  aAddress              A pointer to the IPv6 destination.
+ * @param[in]  aCallback             A pointer to a function called on receiving an Energy Report message.
+ * @param[in]  aContext              A pointer to application-specific context.
+ *
+ * @retval kThreadError_None         Successfully enqueued the PAN ID Query message.
+ * @retval kThreadError_NoBufs       Insufficient buffers to generate a PAN ID Query message.
+ * @retval kThreadError_InvalidState The commissioner is not active
```
Comment (jwhui) [inline include/openthread/commissioner.h]: nit: add period (`.`) at end of line.

Comment (librasungirl) [inline include/openthread/commissioner.h]: Thanks. Updated

**Pull request merged**